bin/coqtop.ml

```diff
@@ -0,0 +1,81 @@
+open Stdune
+open Import
+
+let doc =
+  "Execute the Coq toplevel with the local configuration."
+
+let man =
+  [ `S "DESCRIPTION"
+  ; `P
+      {|$(b,dune coq top FILE -- ARGS) runs the Coq toplevel to process the
+          given $(b,FILE). The given arguments are completed according to the
+          local configuration. This is equivalent to running $(b,coqtop ARGS)
+          with a $(b,_CoqProject) file containing the local configurations
+          from the $(b,dune) files, but does not require maintaining a
+          $(b,_CoqProject) file.|}
+  ; `Blocks Common.help_secs
+  ]
+
+let info = Term.info "top" ~doc ~man
+
+let term =
+  let+ common = Common.term
+  and+ context =
+    Common.context_arg ~doc:{|Run the Coq toplevel in this build context.|}
+  and+ coq_file =
+    Arg.(required & pos 0 (some string) None (Arg.info [] ~docv:"COQFILE"))
+  and+ extra_args =
+    Arg.(value & pos_right 0 string [] (Arg.info [] ~docv:"ARGS"))
+  in
+  let config = Common.init common in
+  (* Checks on the argument file. *)
+  let source = Common.prefix_target common coq_file in
+  if not (Path.is_file (Path.of_string source)) then
+    User_error.raise [ Pp.textf "cannot find file: %s" coq_file ];
+  let coqtop, argv, env =
+    Scheduler.go ~common ~config (fun () ->
+        let open Fiber.O in
+        let* setup = Import.Main.setup () in
+        let* setup = Memo.Build.run setup in
+        let sctx = Import.Main.find_scontext_exn setup ~name:context in
+        let context = Dune_rules.Super_context.context sctx in
+        let dir =
```

Comment:
I suspect there is something wrong with some of the `~dir` arguments I pass around. I don't quite understand what specific directory the functions expect. I think there are three directories of interest:
- the current working directory (from which we want to run `coqtop` (I think),
- the directory containing the file that we want to load (in the source tree, not under `_build`),
- the directory containing the compiled `.vo` files under `_build`.

Reply:
> the directory containing the compiled .vo files under _build.

I think this is the right choice. Which you have taken here as far as I can tell.